playbooks/roles/vault/tasks/main.yml

```diff
@@ -54,16 +54,22 @@
   notify: restart vault
 
 - name: Configure `vault` autocomplete
-  shell: "vault -autocomplete-install"
   become: no
+  shell: "vault -autocomplete-install"
   when: vault_binary.changed
   ignore_errors: yes
 
 - name: Install Vault environment variables
+  become: no
+  template:
+    src: vault-env.j2
+    dest: "/home/{{ ansible_ssh_user }}/.vault-env"
+
+- name: Source Vault environment variables
+  become: no
   lineinfile:
-    path: "$HOME/.bashrc"
-    line: "export {{ item }}"
-  with_items: "{{ vault_env }}"
+    path: "/home/{{ ansible_ssh_user }}/.bashrc"
+    line: "source $HOME/.vault-env"
```

Comment:
I guess this is where Ansible's lack of convergence fails us - now `.bashrc` has both the old `export X=Y` lines plus this new one! It isn't harmful, but it's less than desirable. For now I'll just manually go in and delete those lines.

Reply:
Yeah. I should have provided a cleanup playbook to remove the spurious lines. I'm still very fuzzy regarding when and how the playbooks are applied. Do you do that manually from time to time? With tags? Maybe there is a document I could read explaining this?

Reply:
Yep it's super manual :p I just run it whenever I need to deploy something. I wish we ran them on a cron for all servers regularly, but the truth is some playbooks are _not_ necessarily completely idempotent, in particular some of the external ones. So it's dangerous to run them automatically, but I think long-term that'd be a really important improvement.

I do run with tags, and when I need to be surgical I use `--step`. I used `--step` here to ensure I didn't accidentally restart Vault by accident. Doing that would mean we'd have to unlock it with the unlock keys, and that'll get dirty :)

I'm not sure about docs, I think the beginning part of https://github.com/open-craft/ansible-secrets/ mostly explains it - just need to run `ansible-playbook` with whatever parameters are correct for your goal.